model/map/src/main/java/org/keycloak/models/map/common/EntityField.java

```diff
@@ -48,12 +51,22 @@
      * @param e Entity
      * @param value Value to be added to the collection
      * @throws ClassCastException If this field is not a collection.
+     * @throws UnsupportedOperationException If this collection type is not yet known.
      */
     default <T> void collectionAdd(E e, T value) {
         @SuppressWarnings("unchecked")
         Collection<T> c = (Collection<T>) get(e);
         if (c != null) {
             c.add(value);
+        } else {
+            if (getFieldClass().equals(List.class)) {
+                c = Collections.singletonList(value);
+            } else if (getFieldClass().equals(Set.class)) {
+                c = Collections.singleton(value);
+            } else {
+                throw new UnsupportedOperationException("Unsupported collection type."); // in case we add e.g. java.util.Queue in future
+            }
```

Comment:
this can be simplified to 
```java
c = getFieldClass().equals(List.class) ? Collections.singletonList(value) : Collections.singleton(value);
```
in case we know we'll stick with `Set` and `List` only.